### PR TITLE
rkt: Use volumes from RunContainerOptions

### PR DESCRIPTION
**Pull request by euank**
This replaces the previous creation of mounts from the `volumeGetter`
with mounts provided via RunContainerOptions.

This is motivated by the fact that the latter has a more complete set of
mounts (e.g. the `/etc/hosts` one created in kubelet.go in the case an IP is available).

This does not induce further e2e failures as far as I can tell.

cc @yifan-gu

### CODE REVIEW

**Comment by yifan-gu** [comment]
@euank LGTM
Not sure what test is failing. These days the integration test log is hard to parse :(

**Comment by euank** [comment]
```
--- FAIL: TestCascadingDeletion (130.05s)
	garbage_collector_test.go:211: timed out waiting for the condition
```

@k8s-bot test this issue: #25890

**Comment by euank** [comment]
Hey, lucky me, I hit the same failure

@k8s-bot integration test this issue: #25890

**Comment by euank** [comment]
@k8s-bot unit test this issue: #25890

**Comment by euank** [comment]
@k8s-bot test this issue: #26270 

Hopefully this doesn't make the network quota exhaustion worse.

**Comment by euank** [comment]
@k8s-bot test this issue: #26302

**Comment by euank** [comment]
lost my LGTM rebasing, /cc @vishh since you were originally picked by the bot and @yifan-gu's at reduced latency.

**Comment by vishh** [comment]
LGTM

**Comment by euank** [comment]
Aaand another rebase and lgtm lost (the kubetypes move).

**Comment by euank** [comment]
cc @vishh
